File: frontend/src/model/VideoDto.ts
import type { ChannelDto } from "./ChannelDto";

type VideoDto = {
    id: string;
    channelId: string;
    channel: ChannelDto;
    title: string;
    description: string;
    thumbnailUrl: string;
    commentCount: number;
    replyCount: number;
    totalReplyCount: number;
    commentsDisabled: boolean;
    publishedAt: string;
    lastFetchedAt: string;
};

export type VideoWithChannelIdDto = Omit<VideoDto, "channel">;

export type VideoWithChannelDto = Omit<VideoDto, "channelId">;

File: backend/src/main/java/ca/metricalsky/yt/comments/dto/VideoDto.java
package ca.metricalsky.yt.comments.dto;

import lombok.Data;

import java.time.OffsetDateTime;

@Data
public class VideoDto {

    private String id;

    private String channelId;

    private ChannelDto channel;

    private String title;

    private String description;

    private String thumbnailUrl;

    private Long commentCount;

    private Long replyCount;

    private Long totalReplyCount;

    private OffsetDateTime publishedAt;

    private OffsetDateTime lastFetchedAt;
}
